fast_barcode_scanner/android/src/main/kotlin/com/jhoogstraat/fast_barcode_scanner/Camera.kt

```diff
@@ -89,19 +98,22 @@ class Camera(
             .setBarcodeFormats(0, *scannerConfiguration.formats)
             .build()
 
-        barcodeScanner = MLKitBarcodeScanner(options, { codes ->
-            if (codes.isNotEmpty()) {
-                if (scannerConfiguration.mode == DetectionMode.pauseDetection) {
-                    stopDetector()
-                } else if (scannerConfiguration.mode == DetectionMode.pauseVideo) {
-                    stopCamera()
+        barcodeScanner = MLKitBarcodeScanner(options, object : OnDetectedListener<List<Barcode>> {
+            override fun onSuccess(codes: List<Barcode>, image: Image) {
+                if (codes.isNotEmpty()) {
+                    if (scannerConfiguration.mode == DetectionMode.pauseDetection) {
+                        stopDetector()
+                    } else if (scannerConfiguration.mode == DetectionMode.pauseVideo) {
+                        stopCamera()
+                    }
+                    val base64Image =
+                        imageToBase64(image, Bitmap.CompressFormat.JPEG);
```

Comment:
I'm a bit worried about the impact to performance if we perform compression and conversion for every call to `onSuccess`. I would like to offer another strategy:

1. add the `Image` to a cache with a UUID as a key for each Image cache entry. ***
2. return the Image's cache ID back to Flutter
3. Add a new PlatformChannel method to retrieve an image by UUID. We will look up the image in the cache and perform conversion at that time.

*** The image cache should enforce a maximum size so we don't run out of memory. It should be big enough to hold the image we want, but small enough that we can evict cached Images that we aren't going to use. The App will need to decide if if wants the image associated with the barcode that was read. If it does, it will ask for the image after the barcode value has been received.

Reply:
@dustin-graham Because this function always triggers when the barcode is detected, I use the detected code instead of UUID as a key to save the image. It means that for each code, we only save an image to cache

Reply:
@tnvinh2711 I love that! Great idea.